src/ssl.c

```diff
@@ -47570,6 +47714,43 @@ int wolfSSL_X509_check_ip_asc(WOLFSSL_X509 *x, const char *ipasc,
 }
 #endif
 
+#if defined(OPENSSL_EXTRA) && defined(WOLFSSL_CERT_GEN)
+int wolfSSL_X509_check_email(WOLFSSL_X509 *x, const char *chk, size_t chklen,
+                             unsigned int flags)
+{
+    int         ret;
+    DecodedCert dCert;
+
+    (void)flags;
+
+    WOLFSSL_ENTER("wolfSSL_X509_check_email");
+
+    if ((x == NULL) || (chk == NULL)) {
+        WOLFSSL_MSG("Invalid parameter");
+        return WOLFSSL_FAILURE;
+    }
+
+    InitDecodedCert(&dCert, x->derCert->buffer, x->derCert->length, NULL);
```

Comment:
Would wolfSSL_X509_NAME_get_text_by_NID() get the email address from the WOLFSSL_X509.

Reply:
Yeah, that could work. I'll give it a try and use that if it does.